lib/attrsets.nix

```diff
@@ -523,6 +523,36 @@ rec {
     listToAttrs (map (attr: f attr set.${attr}) (attrNames set));
 
 
+  /* Like mapAttrs, but additionally taking a variable that
+     will be automatically incremented, starting from 0.
+
+     Example:
+       imapAttrs (i: name: value: value + i)
+          { a = 1; b = 1; c = 1; }
+       => { a = 1; b = 2; c = 3; }
+
+     Type:
+       imapAttrs :: (Int -> String -> a -> b) -> { ... :: a } -> { ... :: b }
+  */
+  imapAttrs = f: set:
```

Comment:
I'd also recommend `imap0Attrs` until we get around to changing the behavior of `imap`.

Reply:
Oh damn yeah, I thought imap was already gone because I couldn't find it in `lib/lists.nix`, but turns out it's still in `lib/deprecated.nix`, which is oddly imported as `lib.misc.*`. Yeah considering that, this should definitely be `imap0Attrs` instead then.